src/display/annotation_layer.js

```diff
@@ -556,6 +565,100 @@ class LinkAnnotationElement extends AnnotationElement {
     }
     link.className = "internalLink";
   }
+
+  _bindResetFormAction(link, resetForm) {
+    link.className = "internalLink";
+    const otherClickAction = link.onclick;
+
+    link.onclick = () => {
+      if (otherClickAction) {
+        otherClickAction();
+      }
+
+      const {
+        fields: resetFormFields,
+        refs: resetFormRefs,
+        include,
+      } = resetForm;
+      if (resetFormFields.length !== 0 && !this._fieldsObjects) {
```

Comment:
Sorry, but I really don't understand what the `resetFormFields.length !== 0` check is doing here!?

Below it's e.g. sufficient for `resetFormRefs.length !== 0` in order to reach line https://github.com/mozilla/pdf.js/pull/14083/files#diff-d7b040092cf6be09a276b785ba03c46e79bc8342cf4ae4fb1bc139dea8513425R600 which will throw when the fieldObjects aren't defined.

Again, why not use a straight `if (!this._fieldObjects) {`-check for the early return?